sentinelhub/data_utils.py

```diff
@@ -0,0 +1,166 @@
+"""
+Module with statistics to dataframe transformation.
+"""
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from .exceptions import PandasNotInstalled
+from .time_utils import parse_time
+from .type_utils import JsonDict
+
+
+def _extract_hist(hist_cols: Tuple[str, str], hist_data: List[JsonDict]) -> Dict[str, List[float]]:
+    """Transform Statistical API histogram into pandas.DataFrame entry
+
+    :param hist_cols: Column names to store histogram as a dataframe. Example: ("ndvi_B0_bins", "ndvi_B0_counts").
+    :param hist_data: An input representation of Statistical API histogram.
+    :return: Statistical histogram stored as bins and counts.
+    """
+
+    nbins = len(hist_data)
+    bins, counts = [], []
+    for idx in range(nbins):
+        bins.append(hist_data[idx]["lowEdge"])
+        if idx == nbins - 1:
+            bins.append(hist_data[idx]["highEdge"])
+        counts.append(hist_data[idx]["count"])
+
+    return {hist_cols[0]: bins, hist_cols[1]: counts}
+
+
+def _extract_percentiles(col_name: str, perc_data: Dict[str, float]) -> Dict[str, float]:
+    """Transform statistical percentiles into pandas.DataFrame entry
+
+    :param col_name: Output name and band name. Example: ndvi_B0.
+    :param perc_data: An input representation of percentile data of <col_name>.
+    :return: Percentile data as a pandas.DataFrame entry. Examples: {"ndvi_B0_<perc>": perc_val}.
+    """
+    perc_entry = {}
+    for perc, perc_val in perc_data.items():
+        perc_col_name = f"{col_name}_{perc}"
+        perc_entry[perc_col_name] = perc_val
+    return perc_entry
```

Comment:
could this perhaps be just `return {f"{col_name}_{perc}": value for perc, value in perc_data.items()}` If so this function is not necessary.